lib/std/Target.zig

```diff
@@ -880,6 +881,8 @@ pub fn toElfMachine(target: Target) std.elf.EM {
         .ve,
         .wasm32,
         .wasm64,
+        .propeller1,
+        .propeller2,
         => .NONE,
```

Comment:
https://github.com/parallaxinc/propgcc/blob/b280e7a77103d479e82cc687711de6a45579ad70/binutils/include/elf/common.h#L408-L409

Reply:
I'm also adding a P2/PROPELLER2 target taken from here:
https://github.com/ne75/llvm-project/blob/ee801ce165dc3a968ea78d7f793c713a91aabe97/llvm/include/llvm/BinaryFormat/ELF.h#L322C3-L322C15

Reply:
Hmm... it's a little unfortunate that the LLVM port deviates in a few ways from the precedent established by the GCC port much earlier.

The choice of `EM_P2 = 300` is particularly problematic; ports that haven't been assigned an ELF machine type "officially" are supposed to use a larger number picked at random, like the ones [here](https://github.com/ziglang/zig/blob/3b465ebec59ee942b6c490ada2f81902ec047d7f/lib/std/elf.zig#L1498-L1553), to prevent conflicts.

Further, the binutils port in the propgcc repo differentiates P1 and P2 by [some bits in ELF's `e_flags` field](https://github.com/parallaxinc/propgcc/blob/b280e7a77103d479e82cc687711de6a45579ad70/binutils/include/elf/propeller.h#L58-L65), not by `e_machine`.

Frankly, I think the binutils port is in the right here, and the LLVM port is doing things differently for no good reason that I can tell. So I think we should omit `EM_P2` and just return `EM_PROPELLER` for both cases, and ideally also engage with the maintainer of the LLVM port and see if they'd be willing to change this to conform to the binutils port.